### PR TITLE
ci: ensure "xxx succeeded" properly fail

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

Fixes https://github.com/RIOT-OS/RIOT/issues/20287 and a similar issue for the "tools-build succeeded" ci job.

### CODE REVIEW

**Comment by kaspar030** [comment]
The last fixup commit shows this is now failling this PR b/c of the fixup commit.

**Comment by kaspar030** [comment]
(now that I'm looking at this, the tools-build-success has the same issue)

File: .github/workflows/check-commits.yml [hunk 1]
```diff
@@ -28,8 +28,12 @@ jobs:
...
+    if: always() && github.event_name != 'merge_group'
     runs-on: ubuntu-latest
     steps:
     - name: check-commits succeeded
-      run: exit 0
+      uses: re-actors/alls-green@release/v1
+      with:
+        jobs: ${{ toJSON(needs) }}
+
+## adding some fixup
```
Comment (maribu) [inline .github/workflows/check-commits.yml:39]: I don't get this comment

Comment (kaspar030) [reply .github/workflows/check-commits.yml:39]: that's from the test fixup commit (which makes this PR fail)

**Comment by kaspar030** [comment]
Sorry for the late addition, I've added a similar fix for the tools build test, plus another test commit.

**Comment by kaspar030** [comment]
tools-buildtest failed (as expected due to the test commit that broke edbg)

**Comment by kaspar030** [comment]
Sorry again, but now this is ready.

**Review by maribu — APPROVED** [review]

**Pull request merged**